rdflib/compare.py

```diff
@@ -286,6 +290,9 @@ def _initial_color(self):
                     self._neighbors[s].add(o)
                 if isinstance(o, BNode):
                     self._neighbors[o].add(s)
+                if isinstance(p, BNode):
```

Comment:
i think this should never happen (by RDF spec predicates have to be IRIs), but i guess it doesn't hurt...